File: Python/005LongestPalindrome.py
class Solution:
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        length  = 0
        result = ''
        for i in range(len(s)):
            current_length = self.palindromeLengthSymmetric(s, i,i+1)
            if current_length > length:
                result = s[i-int(current_length/2)+1:i+int(current_length/2)+1]
                length = current_length
            current_length = self.palindromeLengthNonSymmeteric(s,i)
            if current_length > length:
                result = s[i-int((current_length-1)/2):i+int((current_length-1)/2+1)]
                length = current_length
        return result
        
    def palindromeLengthSymmetric(self, s, left, right):
        result = 0
        while left > -1 and right < len(s):
            if s[left] == s[right] :
                left -= 1
                right += 1
                result += 2
            else:
                break
        return result
    
    def palindromeLengthNonSymmeteric(self, s, middle):
        result = 1
        return result + self.palindromeLengthSymmetric(s, middle-1, middle+1)

File: DotNet/005LongestPalindrome.cs
public class Solution {
    public string LongestPalindrome(string s) {
        int length = 0;
        string palindrome="";
        for(int i = 0; i<s.Length; i++){
            int currentLength = PalindromicLength(s, i, i+1);
            if (currentLength > length){
                palindrome=s.Substring(i-currentLength/2+1,currentLength);
                length = currentLength;
            }
            currentLength = PalindromicLength(s, i);
            if(currentLength > length){
                palindrome = s.Substring(i-(currentLength-1)/2, currentLength);
                length = currentLength;
            }
        }
        return palindrome;
        
    }
    private int PalindromicLength(string s, int left, int right){
        int result = 0;
        while(left > -1 && right < s.Length){
            if(s[left]==s[right]){
                left--;
                right++;
                result += 2;
            }else{
                break;
            }
        }
        return result;
    }
    private int PalindromicLength(string s, int middle){
        int result = 1;
        return result + PalindromicLength(s, middle-1, middle+1);
    }
}